packages/compat/src/v1-config.ts

```diff
@@ -49,16 +49,12 @@ export class WriteV1Config extends Plugin {
 }
 
 function metaLoader(appName: string) {
+  // Added contentFor config-module support.
+  // ember-cli adds content in AMD module format and
+  // not as ES6 JS modules. Changed template to match ember-cli
```

Comment:
Although ember-cli produces AMD here, could we continue to produce ES6 modules? Or is there a technical limitation preventing us from doing so.

In general, the idea of AMD in embroider is only for compatibility. If core features such as `config/environment` cannot be implemented in the modern way, maybe embroider is missing an important primitive.